pkg/clusterresource/vsphere.go

```diff
@@ -0,0 +1,125 @@
+package clusterresource
+
+import (
+	"fmt"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	installertypes "github.com/openshift/installer/pkg/types"
+	installervsphere "github.com/openshift/installer/pkg/types/vsphere"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+	hivev1vsphere "github.com/openshift/hive/pkg/apis/hive/v1/vsphere"
+	//"github.com/openshift/hive/pkg/constants"
+)
+
+var _ CloudBuilder = (*VSphereCloudBuilder)(nil)
+
+// VSphereCloudBuilder encapsulates cluster artifact generation logic specific to Vsphere.
+type VSphereCloudBuilder struct {
+	// VCenter is the domain name or IP address of the vCenter.
+	VCenter string `json:"vCenter"`
+
+	// Username is the name of the user to use to connect to the vCenter.
+	Username string `json:"username"`
+
+	// Password is the password for the user to use to connect to the vCenter.
+	Password string `json:"password"`
+
+	// Datacenter is the name of the datacenter to use in the vCenter.
+	Datacenter string `json:"datacenter"`
+
+	// DefaultDatastore is the default datastore to use for provisioning volumes.
+	DefaultDatastore string `json:"defaultDatastore"`
+
+	// Folder is the name of the folder that will be used and/or created for
+	// virtual machines.
+	Folder string `json:"folder,omitempty"`
+
+	// Cluster is the name of the cluster virtual machines will be cloned into.
+	Cluster string `json:"cluster,omitempty"`
+
+	// ClusterOSImage overrides the url provided in rhcos.json to download the RHCOS OVA
+	ClusterOSImage string `json:"clusterOSImage,omitempty"`
+
+	// APIVIP is the virtual IP address for the api endpoint
+	APIVIP string `json:"apiVIP,omitempty"`
+
+	// IngressVIP is the virtual IP address for ingress
+	IngressVIP string `json:"ingressVIP,omitempty"`
+
+	// DNSVIP is the virtual IP address for DNS
+	DNSVIP string `json:"dnsVIP,omitempty"`
+
+	// DefaultMachinePlatform is the default configuration used when
+	// installing on VSphere for machine pools which do not define their own
+	// platform configuration.
+	// +optional
+	DefaultMachinePlatform *hivev1vsphere.MachinePool `json:"defaultMachinePlatform,omitempty"`
+
+	// Network specifies the name of the network to be used by the cluster.
+	Network string `json:"network,omitempty"`
+}
+
+func (p *VSphereCloudBuilder) generateCredentialsSecret(o *Builder) *corev1.Secret {
+	return &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      p.credsSecretName(o),
+			Namespace: o.Namespace,
+		},
+		Type: corev1.SecretTypeOpaque,
+		StringData: map[string]string{
+			"govc_host":     p.VCenter,
```

Comment:
Do we need the "govc_" clarification? Or will "host", "username", and "password" suffice?

Reply:
I don't think we need them.